src/main/java/uk/ac/cam/cl/dtg/isaac/quiz/IsaacNumericValidator.java

```diff
@@ -513,11 +513,25 @@ private QuantityValidationResponse exactStringMatch(final IsaacNumericQuestion i
                                                         final Quantity answerFromUser) {
 
         log.debug("\t[exactStringMatch]");
+        boolean wasACorrectAnswerWithUsersSelectedUnit = false;
 
-        for (Choice c : isaacNumericQuestion.getChoices()) {
+        // Goes through correct answers first, in order to work out whether any correct answers match the units that
+        // the user selected.
+        List<Choice> orderedChoices = getOrderedChoices(isaacNumericQuestion.getChoices());
```

Comment:
Doing this sorting is wasteful, since we don't ever _use_ the result of `exactStringMatch` unless the answer is a match to a known incorrect choice; for every other case (i.e. the vast majority), we then do the normal `validateWith[out]Units` which does this sorting again.

I see that in order for the `wasACorrectAnswerWithUsersSelectedUnit` variable to be useful we need this sorting, but I wonder at this point if it is time to just remove the `exactStringMatch` check entirely - since it was only added to be a very quick very simple short circuit long before most of the rest of the complexity was added on.